File: app/src/main/java/com/spacemonster/webdataviewer/view/webimage/WebImageViewHolder.java
package com.spacemonster.webdataviewer.view.webimage;

import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.spacemonster.webdataviewer.R;
import com.spacemonster.webdataviewer.view.AbstractDataViewHolder;
import com.spacemonster.webdataviewer.content.data.WebImage;
import com.squareup.picasso.Picasso;

/**
 * Created by GE62 on 2017-12-21.
 */

public class WebImageViewHolder extends AbstractDataViewHolder<WebImage> {
    private static final boolean USE_GLIDE = true;

    private ImageView urlView;
    private RequestOptions requestOptions;

    public WebImageViewHolder(View itemView) {
        super(itemView);
        urlView = itemView.findViewById(R.id.iv_url_img);
        //Use
        if(USE_GLIDE == true){
            requestOptions = new RequestOptions();
            requestOptions.placeholder(R.drawable.wait).centerInside().fitCenter();
        }
    }

    @Override
    public void initView(WebImage data) {
        if(USE_GLIDE == true){
            Glide.with(itemView.getContext()).asDrawable().load(WebImage.MAIN_URL + data.getImageUrl()).apply(requestOptions).into(urlView);
        }else{
            Picasso.with(itemView.getContext()).load(WebImage.MAIN_URL + data.getImageUrl()).fit().centerInside().placeholder(R.drawable.wait).into(urlView);
        }
    }
}

File: app/src/main/kotlin/com/spacemonster/webdataviewer/view/webimage/WebImageViewHolder.kt
package com.spacemonster.webdataviewer.view.webimage

import android.view.View
import android.widget.ImageView

import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.spacemonster.webdataviewer.R
import com.spacemonster.webdataviewer.view.AbstractDataViewHolder
import com.spacemonster.webdataviewer.content.data.WebImage
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.item_image.view.*

/**
 * Created by GE62 on 2017-12-21.
 */

class WebImageViewHolder(itemView: View) : AbstractDataViewHolder<WebImage>(itemView) {

    private var requestOptions: RequestOptions = RequestOptions()

    init {
        //Use
        if (USE_GLIDE) {
            requestOptions.placeholder(R.drawable.wait).centerInside().fitCenter()
        }
    }

    override fun initView(data: WebImage) {
        if (USE_GLIDE) {
            Glide.with(itemView.context).asDrawable().load(WebImage.MAIN_URL + data.imageUrl).apply(requestOptions!!).into(itemView.iv_url_img)
        } else {
            Picasso.with(itemView.context).load(WebImage.MAIN_URL + data.imageUrl).fit().centerInside().placeholder(R.drawable.wait).into(itemView.iv_url_img)
        }
    }

    companion object {
        private val USE_GLIDE = true
    }
}
